generators.cpp

```diff
@@ -0,0 +1,331 @@
+#define PI 3.14159265358979323846
```

Comment:
Avoid using `define`, use const instead.
`const double PI = 3.14159265358979323846`

`define` just inlines the value instead of the `PI` in your code at compile time without any checks, while if you use const, you'll get type checks and compile errors in case you use it wrong.

Also, move it down after the `include`s section. Put it after a newline between `using namespace std` and the first func.